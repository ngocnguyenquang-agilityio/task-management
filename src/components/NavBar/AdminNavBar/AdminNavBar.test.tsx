import { render, screen } from '@testing-library/react';
import { AdminNavBar } from '.';

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  usePathname: () => 'localhost:3000/example/path',
}));

describe('AdminNavBar component', () => {
  test('renders the AdminNavBar match snapshot', () => {
    const { container } = render(<AdminNavBar />);
    expect(container).toMatchSnapshot();
  });

  test('renders the AdminNavBar with children', () => {
    render(<AdminNavBar />);
    expect(screen.getByText('Boards')).toBeInTheDocument();
  });
});
