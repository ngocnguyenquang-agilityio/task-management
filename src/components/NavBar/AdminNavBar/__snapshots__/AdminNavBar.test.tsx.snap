// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavBar component renders the AdminNavBar match snapshot 1`] = `
<div>
  <nav
    class="relative lg:static w-full"
  >
    <div
      class="flex items-center justify-between py-[10px] pr-5 lg:pr-0 px-[5px] lg:mb-[31px] rounded-lg bg-white dark:bg-zinc-800 "
    >
      <a
        class="flex items-center gap-2 w-full"
        href="/admin/boards"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 25 26"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="24.5357"
            rx="7.37141"
            width="24.5357"
            y="0.732178"
          />
          <path
            d="M12.8821 20.6785C11.8199 20.6785 10.8217 20.4768 9.88747 20.0734C8.95325 19.6701 8.1406 19.1231 7.44953 18.4325C6.75846 17.7415 6.21149 16.9288 5.80862 15.9946C5.40575 15.0604 5.20406 14.0622 5.20355 13C5.20355 11.9378 5.40524 10.9396 5.80862 10.0053C6.212 9.0711 6.75897 8.25846 7.44953 7.56739C8.1406 6.87632 8.95325 6.32935 9.88747 5.92648C10.8217 5.52361 11.8199 5.32192 12.8821 5.32141C13.4964 5.32141 14.0948 5.3918 14.6774 5.53257C15.2599 5.67334 15.8197 5.87811 16.3567 6.14686C16.5486 6.24924 16.6735 6.40281 16.7314 6.60757C16.7892 6.81233 16.7539 7.00429 16.6254 7.18346C16.4974 7.36263 16.3277 7.4778 16.1163 7.529C15.9049 7.58019 15.6971 7.55459 15.4928 7.45221C15.0833 7.26025 14.6576 7.11307 14.2159 7.01069C13.7741 6.90831 13.3295 6.85712 12.8821 6.85712C11.18 6.85712 9.73057 7.45554 8.53374 8.65237C7.33691 9.8492 6.73875 11.2984 6.73926 13C6.73926 14.702 7.33768 16.1515 8.53451 17.3483C9.73134 18.5452 11.1805 19.1433 12.8821 19.1428C14.5842 19.1428 16.0336 18.5446 17.2305 17.3483C18.4273 16.152 19.0255 14.7026 19.0249 13C19.0249 12.8976 19.0219 12.7985 19.0157 12.7028C19.0096 12.6071 18.9999 12.5078 18.9866 12.4049C18.961 12.1873 19.0027 11.9795 19.1117 11.7814C19.2208 11.5833 19.3838 11.452 19.6008 11.3875C19.8056 11.3235 19.9976 11.3427 20.1767 11.4451C20.3559 11.5474 20.4583 11.701 20.4839 11.9058C20.5095 12.0849 20.5287 12.2641 20.5415 12.4433C20.5543 12.6224 20.5607 12.808 20.5607 13C20.5607 14.0622 20.359 15.0604 19.9556 15.9946C19.5522 16.9288 19.0052 17.7415 18.3147 18.4325C17.6236 19.1236 16.811 19.6708 15.8767 20.0742C14.9425 20.4776 13.9443 20.679 12.8821 20.6785Z"
          />
          <path
            d="M18.9482 7.22192L11.8071 14.3822L10.1562 12.7313C10.0154 12.5905 9.83627 12.5201 9.61872 12.5201C9.40116 12.5201 9.22199 12.5905 9.08122 12.7313C8.94044 12.8721 8.87006 13.0512 8.87006 13.2688C8.87006 13.4863 8.94044 13.6655 9.08122 13.8063L11.2696 15.9947C11.4232 16.1482 11.6023 16.225 11.8071 16.225C12.0119 16.225 12.191 16.1482 12.3446 15.9947L20.0232 8.29692C20.1639 8.15615 20.2343 7.97698 20.2343 7.75942C20.2343 7.54186 20.1639 7.3627 20.0232 7.22192C19.8701 7.06784 19.6879 6.99412 19.4764 7.00078C19.265 7.00743 19.0889 7.08115 18.9482 7.22192Z"
          />
        </svg>
        <h1
          class="text-xl font-bold dark:text-white"
        >
          Taskboard
        </h1>
      </a>
      <svg
        class="lg:hidden cursor-pointer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Menu_Burger"
        >
          <path
            d="M3.563,4.063c-0.276,-0 -0.5,-0.224 -0.5,-0.5c-0,-0.276 0.224,-0.5 0.5,-0.5l16.874,-0.001c0.276,-0 0.5,0.224 0.5,0.5c0,0.276 -0.224,0.5 -0.5,0.5l-16.874,0.001Z"
          />
          <path
            d="M3.563,12.501c-0.276,-0 -0.5,-0.224 -0.5,-0.5c-0,-0.276 0.224,-0.5 0.5,-0.5l16.874,-0.002c0.276,0 0.5,0.224 0.5,0.5c0,0.276 -0.224,0.5 -0.5,0.5l-16.874,0.002Z"
          />
          <path
            d="M3.563,20.939c-0.276,-0 -0.5,-0.224 -0.5,-0.5c-0,-0.276 0.224,-0.5 0.5,-0.5l16.874,-0.002c0.276,0 0.5,0.224 0.5,0.5c0,0.276 -0.224,0.5 -0.5,0.5l-16.874,0.002Z"
          />
        </g>
      </svg>
    </div>
    <div
      class="mt-2 rounded-lg bg-white dark:bg-zinc-800 px-3 py-4 w-full absolute lg:static lg:block z-50 hidden"
    >
      <span
        class="text-xs pb-2 font-bold text-zinc-500 dark:text-white "
      >
        Menu
      </span>
      <div
        class="flex flex-col gap-4 mt-2"
      >
        <a
          class="flex gap-2 items-center py-2 px-4 rounded-lg text-zinc-500 fill-zinc-500 dark:text-white dark:fill-white hover:bg-black hover:text-white hover:fill-white hover:font-bold"
          href="/admin/boards"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.91669 1.6665C7.58517 1.6665 7.26722 1.7982 7.0328 2.03262C6.79838 2.26704 6.66669 2.58498 6.66669 2.9165V3.74984C6.66669 4.08136 6.79838 4.3993 7.0328 4.63372C7.26722 4.86814 7.58517 4.99984 7.91669 4.99984H12.0834C12.4149 4.99984 12.7328 4.86814 12.9672 4.63372C13.2017 4.3993 13.3334 4.08136 13.3334 3.74984V2.9165C13.3334 2.58498 13.2017 2.26704 12.9672 2.03262C12.7328 1.7982 12.4149 1.6665 12.0834 1.6665H7.91669Z"
            />
            <path
              clip-rule="evenodd"
              d="M3.2325 4.06426C3.70667 3.58926 4.36833 3.42259 5.41667 3.36426V3.75009C5.41667 4.41313 5.68006 5.04902 6.1489 5.51786C6.61774 5.9867 7.25363 6.25009 7.91667 6.25009H12.0833C12.7464 6.25009 13.3823 5.9867 13.8511 5.51786C14.3199 5.04902 14.5833 4.41313 14.5833 3.75009V3.36426C15.6317 3.42259 16.2933 3.58926 16.7675 4.06426C17.5 4.79676 17.5 5.97509 17.5 8.33176V13.3318C17.5 15.6893 17.5 16.8676 16.7675 17.6001C16.0358 18.3318 14.8567 18.3318 12.5 18.3318H7.5C5.14333 18.3318 3.96417 18.3318 3.2325 17.6001C2.5 16.8668 2.5 15.6893 2.5 13.3318V8.33176C2.5 5.97509 2.5 4.79676 3.2325 4.06426ZM10.625 9.16676C10.625 9.001 10.5592 8.84203 10.4419 8.72482C10.3247 8.60761 10.1658 8.54176 10 8.54176C9.83424 8.54176 9.67527 8.60761 9.55806 8.72482C9.44085 8.84203 9.375 9.001 9.375 9.16676V11.0418H7.5C7.33424 11.0418 7.17527 11.1076 7.05806 11.2248C6.94085 11.342 6.875 11.501 6.875 11.6668C6.875 11.8325 6.94085 11.9915 7.05806 12.1087C7.17527 12.2259 7.33424 12.2918 7.5 12.2918H9.375V14.1668C9.375 14.3325 9.44085 14.4915 9.55806 14.6087C9.67527 14.7259 9.83424 14.7918 10 14.7918C10.1658 14.7918 10.3247 14.7259 10.4419 14.6087C10.5592 14.4915 10.625 14.3325 10.625 14.1668V12.2918H12.5C12.6658 12.2918 12.8247 12.2259 12.9419 12.1087C13.0592 11.9915 13.125 11.8325 13.125 11.6668C13.125 11.501 13.0592 11.342 12.9419 11.2248C12.8247 11.1076 12.6658 11.0418 12.5 11.0418H10.625V9.16676Z"
              fill-rule="evenodd"
            />
          </svg>
          Boards
        </a>
        <a
          class="flex gap-2 items-center py-2 px-4 rounded-lg text-zinc-500 fill-zinc-500 dark:text-white dark:fill-white hover:bg-black hover:text-white hover:fill-white hover:font-bold"
          href="/admin/projects"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.3334 7.50004C18.3334 7.50004 17.5059 8.24171 15.8509 8.90421L13.5109 9.84087C11.8559 10.5025 11.0275 10.8334 10 10.8334C8.97169 10.8334 8.14419 10.5025 6.48919 9.84087L4.15002 8.90421C2.49419 8.24254 1.66669 7.50004 1.66669 7.50004M1.66669 10.8334C1.66669 10.8334 2.49419 11.575 4.14919 12.2375L6.48919 13.1742C8.14419 13.8359 8.97252 14.1667 10 14.1667C10.795 14.1667 11.47 13.9692 12.5 13.5734M15.8509 12.2375C17.5059 11.5759 18.3334 10.8334 18.3334 10.8334M4.14919 2.76254C2.49419 3.42421 1.66669 3.75504 1.66669 4.16671C1.66669 4.57837 2.49419 4.90837 4.14919 5.57087L6.48919 6.50754C8.14419 7.16921 8.97252 7.50004 10 7.50004C11.0284 7.50004 11.8559 7.16921 13.5109 6.50754L15.8509 5.57087C17.5059 4.90921 18.3334 4.57837 18.3334 4.16671C18.3334 3.75504 17.5059 3.42504 15.8509 2.76254L13.5109 1.82671C11.8559 1.16421 11.0275 0.833374 10 0.833374C9.20502 0.833374 8.53002 1.03087 7.50002 1.42671"
              stroke-linecap="round"
              stroke-width="1.5"
            />
          </svg>
          Projects
        </a>
        <a
          class="flex gap-2 items-center py-2 px-4 rounded-lg text-zinc-500 fill-zinc-500 dark:text-white dark:fill-white hover:bg-black hover:text-white hover:fill-white hover:font-bold"
          href="/admin/tasks"
        >
          <svg
            class="w-4 h-4"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 1H3C1.89543 1 1 1.89543 1 3V17C1 18.1046 1.89543 19 3 19H17C18.1046 19 19 18.1046 19 17V3C19 1.89543 18.1046 1 17 1Z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M7 6H14M6 10H12M9 14H14"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
          Tasks
        </a>
        <button
          class="flex items-center rounded-lg p-2 disabled:cursor-not-allowed border border-gray-200 bg-transparent dark:border-white dark:hover:bg-zinc-100 dark:hover:text-black gap-2 px-4 hover:bg-black hover:text-white hover:fill-white hover:font-bold text-zinc-500 fill-zinc-500 dark:text-white dark:fill-white border-none"
          type="button"
        >
          <div
            class=""
          >
            <svg
              class="w-4 h-4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
          </div>
          Sign Out
        </button>
      </div>
      <div
        class="mt-2"
      >
        <span
          class="text-xs font-bold text-zinc-500 dark:text-white"
        >
          Theme
        </span>
        <button
          class="flex items-center rounded-lg p-2 disabled:cursor-not-allowed border border-gray-200 bg-transparent dark:border-white dark:text-white w-full mt-2 hover:bg-black hover:text-white dark:hover:bg-black dark:hover:text-white border-none text-zinc-500"
          type="button"
        >
          <div
            class="px-2"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M14 4c.34 0 .68.02 1.01.07C13.1 6.23 12 9.05 12 12s1.1 5.77 3.01 7.93c-.33.05-.67.07-1.01.07-4.41 0-8-3.59-8-8s3.59-8 8-8m0-2C8.48 2 4 6.48 4 12s4.48 10 10 10c1.82 0 3.53-.5 5-1.35-2.99-1.73-5-4.95-5-8.65s2.01-6.92 5-8.65A9.973 9.973 0 0 0 14 2z"
              />
            </svg>
          </div>
          <span>
            Dark
          </span>
        </button>
      </div>
    </div>
  </nav>
</div>
`;
